compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -716,12 +716,18 @@ class TreeUnpickler(reader: TastyReader,
             }
             TypeDef(readTemplate(localCtx))
           } else {
+            sym.info = TypeBounds.empty // needed to avoid cyclic references when unpicklin rhs, see i3816.scala
+            sym.setFlag(Provisional)
             val rhs = readTpt()(localCtx)
-            sym.info = NoCompleter
+            sym.info = new NoCompleter {
+              override def completerTypeParams(sym: Symbol)(implicit ctx: Context) =
+                rhs.tpe.typeParams
+            }
             sym.info = rhs.tpe match {
               case _: TypeBounds | _: ClassInfo => checkNonCyclic(sym, rhs.tpe, reportErrors = false)
               case _ => TypeAlias(rhs.tpe)
             }
+            sym.resetFlag(Provisional)
```

Comment:
how come we don't need to do something similar to `ensureUpToDate` in `Namer#typeDefSig` ?

Reply:
Frankly: I don't know. The other question is why we need to use `abstracted` in Namer but the same example passes OK in `TreeUnpickler`. I tried to remove `abstracted` in Namer, bu then i3816.scala would fail to compile. So the two cases are not the same. But I did not have the time to dig deeper.